### PR TITLE
Move miniKindOf out of if scope to fix ES5 compatibility issue

### PR DESCRIPTION
**Pull request by embeddedt**
---
name: :bug: Bug fix or new feature
about: Fixing a problem with Redux
---

### Does this PR add a new _feature_, or fix a _bug_?

It fixes a bug.

### Why should this PR be included?

Currently redux cannot be run in ES5 environments even if Babel is used.

## Checklist

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Is there an existing issue for this PR?
  - #4089
- [x] Have the files been linted and formatted?
- [x] Have the docs been updated to match the changes in the PR?
- [x] Have the tests been updated to match the changes in the PR?
- [x] Have you run the tests locally to confirm they pass?

### What is the current behavior, and the steps to reproduce the issue?

Please refer to #4089.

**TL;DR** There are functions declared inside an `if` statement, which ES5 forbids and Babel does not fix.

### What is the expected behavior?

Redux should run without any syntax errors, if it's being bundled and transpiled in the standard manner for ES5.

### How does this PR fix the problem?

It moves the aforementioned functions out of the if statement and into the file's scope.

The diff looks huge but it's mostly due to the indentation changing.

### CODE REVIEW

**Comment by markerikson** [comment]
FYI, this really needs to be based against the `4.x` branch if we're going to release it.  `master` is an unreleased TypeScript conversion.

**Comment by embeddedt** [comment]
> FYI, this really needs to be based against the 4.x branch if we're going to release it. master is an unreleased TypeScript conversion.

Thanks; I've force pushed a version based off 4.x.

> Also, I'd like to make sure that the function only actually exists in dev mode to save on byte size, but that itself requires an `if(process.env.NODE_ENV === 'development')` check. Not sure if there's a good way to resolve that.

Assuming that production environments are using something like terser, wouldn't it automatically see the function as dead code and remove it, since the `NODE_ENV` check will be replaced by `false`?

I've tested it locally by replacing the check with `'production' !== 'production'` and running `npx terser --module -c dead_code src/utils/kindOf.js` and it does get eliminated.

**Comment by markerikson** [comment]
Okay, sounds good.  There's a couple other little bits that probably need to go into a 4.1.1, so I'll see if I can get this out in the next couple days.
